modular_skyrat/modules/ashwalkers/code/buildings/railroad.dm

```diff
@@ -0,0 +1,127 @@
+/obj/item/stack/rail_track
+	name = "railroad track"
+	desc = "A primitive form of transportation. Place on any floor to start building a railroad."
+	icon = 'modular_skyrat/modules/ashwalkers/icons/railroad.dmi'
+	icon_state = "rail_item"
+
+/obj/item/stack/rail_track/five
+	amount = 5
+
+/obj/item/stack/rail_track/fifty
+	amount = 50
+
+/obj/item/stack/rail_track/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!isopenturf(target))
+		return ..()
+	if(!proximity_flag)
+		return ..()
+	var/obj/structure/railroad/check_rail = locate() in get_turf(target)
+	if(check_rail)
+		return ..()
+	if(!use(1))
+		return ..()
```

Comment:
I feel like those if() could be condensed into two, instead of four, it looks like a bit of a mess when there's so many of them like that, I feel like.